common/src/main/java/org/apache/nemo/common/dag/DAG.java

```diff
@@ -97,6 +97,14 @@ public V getVertexById(final String id) {
     throw new IllegalVertexOperationException("There is no vertex of id: " + id);
   }
 
+  @Override
+  public E getEdgeById(final String id) {
```

Comment:
As I think this method (and `getEdges()`) could be renamed to represent that it's about IncomingEdge.

Reply:
the incomingEdge variable simply works as a bookkeeper for all edges with the information about which vertex that it's an incoming edge of. So, it's not really about incoming edges in that case. What do you think?

Reply:
Hmm.. users may confuses it with outgoing edges.
Anyway, I'm okay.